datasimulator/utils.py

```diff
@@ -0,0 +1,37 @@
+import random
+
+
+def is_mixed_type(arr):
+    if arr:
+        return any(not isinstance(e, type(arr[0])) for e in arr)
+    return False
+
+
+def random_choice(arr):
+    if arr:
+        return random.choice(arr)
+    return None
+
+
+def is_primitive_type(data):
+    return isinstance(data, (int, float, str))
+
+
+def get_recursive_keys(dictionary):
+    keys = []
+    for key, value in dictionary.items():
+        keys.append(key)
+        if type(value) is dict:
+            keys = keys + get_recursive_keys(value)
+
+    return keys
+
+
+def get_recursive_values(dictionary):
```

Comment:
you _could_ do this non-recursively :slightly_smiling_face: 
```python
from functools import reduce

def get_field(d, fields):
    return reduce(lambda acc, field: acc.get(field, {}), fields, d)

def flatten_dict(d):
    result = []
    to_visit = [[key] for key in d.keys()]
    while to_visit:
        key = to_visit.pop()
        result.append(key[-1])
        value = get_field(d, key)
        if isinstance(value, dict):
            to_visit.extend([key + [next_key] for next_key in value.keys()])
    return result
```

note that with either implementation, the behavior is to return duplicate keys; so you'll get stuff like this:
```
In [1]: d = {'a': {'b': {'c': 1}}, 'b': 4, 'c': 5}

In [2]: flatten_dict(d)
Out[2]: ['b', 'c', 'a', 'b', 'c']
```
(just to make sure this is intended)

Reply:
non-recursive again. Thanks